main/main.c

```diff
@@ -139,13 +141,33 @@ run_signal_handler(void)
 	return ret;
 }
 
+struct log_ratelimit_state *
+get_log_ratelimit_state_by_lcore(void)
+{
+	return &log_ratelimit_states[rte_lcore_id()];
+}
+
+static void
+log_ratelimit_init(void)
+{
```

Comment:
We don't need `log_ratelimit_init()` because the variable `log_ratelimit_states` is global, so it's automatically initialized with zeros, and `log_ratelimit_allow()` ignores states whose field `interval` is zero. Each block will set its own rate limit. Until then, there should be no rate limit.